### PR TITLE
TF-2040 Add calendar date icon in email view

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2040 

## Dependent

- Need merged: https://github.com/linagora/jmap-dart-client/pull/70

## Resolved

- Mobile:

https://github.com/linagora/tmail-flutter/assets/80730648/f32eaa70-4d78-48a5-8edd-3b8e2d48ae58

- Tablet:

https://github.com/linagora/tmail-flutter/assets/80730648/1fd97682-69de-4250-91b1-04670baa7b57

- Responsive browser:

https://github.com/linagora/tmail-flutter/assets/80730648/ee747f2a-c3c0-4100-a58f-72f0fc41c3d7

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

File: lib/features/email/presentation/extensions/calendar_event_extension.dart [hunk 1]
```diff
@@ -175,4 +177,42 @@ extension CalendarEventExtension on CalendarEvent {
       return Colors.transparent;
     }
   }
+
+  DateTime? get localStartDate => startUtcDate?.value.toLocal();
+
+  String get monthStartDateAsString {
```
Comment (sherlockvn) [inline lib/features/email/presentation/extensions/calendar_event_extension.dart:183]: is return type as `String?` better or `String`, wdyt ?

File: lib/features/email/presentation/extensions/calendar_event_extension.dart [hunk 2]
```diff
@@ -175,4 +177,42 @@ extension CalendarEventExtension on CalendarEvent {
...
+    if (localStartDate != null) {
+      return date_format.formatDate(
+        localStartDate!,
+        [date_format.M],
+        locale: AppUtils.getCurrentDateLocale()
+      );
+    } else {
+      return '';
```
Comment (sherlockvn) [inline lib/features/email/presentation/extensions/calendar_event_extension.dart:191]: then you can remove this line

File: lib/features/email/presentation/widgets/calendar_event_information_widget.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  final CalendarEvent calendarEvent;
+
+  const CalendarEventInformationWidget({
+    super.key,
+    required this.calendarEvent
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final responsiveUtils = Get.find<ResponsiveUtils>();
```
Comment (sherlockvn) [inline lib/features/email/presentation/widgets/calendar_event_information_widget.dart:21]: IMO, you should move out of build method

File: lib/features/email/presentation/widgets/calendar_event_information_widget.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                ),
+              )
+            ],
+          )
+        : Row(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Container(
+                padding: const EdgeInsets.all(CalendarEventInformationWidgetStyles.calendarDateIconMargin),
+                color: Colors.black.withOpacity(0.04),
```
Comment (sherlockvn) [inline lib/features/email/presentation/widgets/calendar_event_information_widget.dart:55]: what if it's in dark mode

File: lib/features/email/presentation/extensions/calendar_event_extension.dart (hunk 1 above)
Comment (dab246) [reply lib/features/email/presentation/extensions/calendar_event_extension.dart:183]: We want this value to be non-null.

File: lib/features/email/presentation/widgets/calendar_event_information_widget.dart (hunk 3 above)
Comment (dab246) [reply lib/features/email/presentation/widgets/calendar_event_information_widget.dart:21]: It is a stateless widget, we can initialize the variables in the build function. If you initialize outside the stateless this widget will not have `const constructor`

File: lib/features/email/presentation/widgets/calendar_event_information_widget.dart (hunk 4 above)
Comment (dab246) [reply lib/features/email/presentation/widgets/calendar_event_information_widget.dart:55]: we don't support at this time

Comment (sherlockvn) [reply lib/features/email/presentation/widgets/calendar_event_information_widget.dart:55]: so, IMO, you should put it in style file, so in the future if support you can change it easily

Comment (dab246) [reply lib/features/email/presentation/widgets/calendar_event_information_widget.dart]: It really doesn't matter at this point. But we can make it clearer for the reader. :)

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**